driver/src/deposit_driver.rs

```diff
@@ -0,0 +1,160 @@
+	let snapp_base: serde_json::Value = serde_json::from_str(&contents)?;
+	let snapp_base_abi: String = snapp_base.get("abi").unwrap().to_string();
+
+	// let snapp_address: String = env::var("SNAPP_CONTRACT_ADDRESS")?;
+	// TO do: use snapp_address in next line
+	let address: Address = Address::from("0xC89Ce4735882C9F0f0FE26686c53074E09B0D550");
+	let contract = Contract::from_json(web3.eth(), address, snapp_base_abi.as_bytes())?;
+	// get current state
+	let result = contract.query("getCurrentStateRoot", (), None, Options::default(), None);
+	let curr_state_root: H256 = result.wait()?;
+	let mut state = db_interface::get_current_balances(db_instance.clone(), curr_state_root.clone())?;
+	let accounts = web3.eth().accounts().wait()?;
+
+	// check that operator has sufficient ether
+	let balance = web3.eth().balance(accounts[0], None).wait()?;
+	if balance.is_zero() {
+		panic!("Not sufficient balance for posting updates into the chain");
+	}
+
+	//get depositSlot
+	//
+	let result = contract.query("depositIndex", (), None, Options::default(), None);
+	let current_deposit_ind: U256 = result.wait().expect("Could not get deposit_slot");
+
+	// get latest non-applied deposit_index
+	let mut deposit_ind: i32 = current_deposit_ind.low_u32() as i32 + 1;
+	println!("Current top deposit_slot is {:?}", deposit_ind);
+	let mut found: bool = false;
+
+	// Starting from the last depositSlot, we search backwards to the first non-applied deposit
+	while !found {
+		deposit_ind = deposit_ind - 1;
+		let result = contract.query(
+			"hasDepositBeenApplied",
+			U256::from(deposit_ind),
+			None,
+			Options::default(),
+			None,
+		);
+		found = result.wait()?;
+		if deposit_ind == 0 {
+			break;
+		}
+	}
+	if found {
+		deposit_ind = deposit_ind + 1;
+	}
+	println!("Current pending deposit_slot is {:?}", deposit_ind);
+
+	let result = contract.query(
+		"getDepositCreationBlock",
+		U256::from(deposit_ind),
+		None,
+		Options::default(),
+		None,
+	);
+	let current_deposit_ind_block: U256 = result.wait()?;
+
+	let current_block = web3.eth().block_number().wait()?;
+
+	let result = contract.query(
+		"getDepositHash",
+		U256::from(deposit_ind),
+		None,
+		Options::default(),
+		None,
+	);
+	let deposit_slot_empty_hash: H256 = result.wait()?;
+	let deposit_slot_empty = deposit_slot_empty_hash == H256::zero();
+
+	println!(
+		"Current block is {:?} and the last deposit_ind_creationBlock is {:?}",
+		current_block, current_deposit_ind_block
+	);
+
+	// if 20 blocks have past since the first deposit and we are not in the newest slot, we apply the deposit.
+	if current_deposit_ind_block + 20 < current_block
+		&& deposit_ind != current_deposit_ind.low_u32() as i32 + 1
+	{
+		println!("Next deposit_slot to be processed is {}", deposit_ind);
+		let deposits = db_interface::get_deposits_of_slot(db_instance.clone(), deposit_ind)?;
+		println!("Amount of deposits to be processed{:?}", deposits.len());
+		//rehash deposits
```

Comment:
consider extracting this logic into a small helper method `rehash_deposits`, than you have documentation that is unlikely to get outdated for free and you top level method is shorter and easier to read.